File: app/src/main/java/com/test/kabak/openweather/core/network/CurrentWeatherResponse.java
package com.test.kabak.openweather.core.network;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class CurrentWeatherResponse {
    public ArrayList<WeatherObject> weather;
    public MainObject main;
    public WindObject wind;

    public static class WeatherObject {
        public String description;
        public String icon;
    }

    public static class MainObject {
        @SerializedName("temp_min")
        public float tempMin;

        @SerializedName("temp_max")
        public float tempMax;
    }

    public static class WindObject {
        @SerializedName("speed")
        public int windSpeed;
    }
}

File: app/src/main/java/com/test/kabak/openweather/core/network/dataClasses/CurrentWeatherResponse.kt
package com.test.kabak.openweather.core.network.dataClasses

import com.google.gson.annotations.SerializedName

class CurrentWeatherResponse(
    @SerializedName("weather") val weather: List<WeatherObject>,
    @SerializedName("main") val main: MainObject,
    @SerializedName("wind") val wind: WindObject,
) {
    class WeatherObject(
        @SerializedName("description") var description: String,
        @SerializedName("icon") var icon: String,
    )

    class MainObject(
        @SerializedName("temp_min") val tempMin: Float,
        @SerializedName("temp_max") val tempMax: Float,
    )

    class WindObject(
        @SerializedName("speed") val windSpeed: Float,
    )
}